src/quic/node_quic_buffer-inl.h

```diff
@@ -0,0 +1,120 @@
+#ifndef SRC_NODE_QUIC_BUFFER_INL_H_
+#define SRC_NODE_QUIC_BUFFER_INL_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_quic_buffer.h"
+#include "util.h"
+#include "uv.h"
+
+namespace node {
+
+namespace quic {
+
+void QuicBufferChunk::default_done(int status) {}
+
+QuicBufferChunk::QuicBufferChunk(size_t len)
+    : data_buf_(len),
+      buf_(uv_buf_init(reinterpret_cast<char*>(data_buf_.out()), len)),
+      done_called_(true) {}
+
+QuicBufferChunk::QuicBufferChunk(uv_buf_t buf_, done_cb done)
+    : buf_(buf_) {
+  if (done != nullptr)
+    done_ = std::move(done);
+}
+
+QuicBufferChunk::~QuicBufferChunk() {
+  CHECK(done_called_);
+}
+
+void QuicBufferChunk::Done(int status) {
+  if (done_called_) return;
+  done_called_ = true;
+  if (done_ != nullptr)
+    std::move(done_)(status);
+}
+
+void QuicBufferChunk::MemoryInfo(MemoryTracker* tracker) const {
```

Comment:
Fwiw, this can also go into the `.cc` file, it’s never used as an inline function, and it removes the need to include `memory_tracker-inl.h` when including this file